src/rust/bitbox02-rust-c/src/lib.rs

```diff
@@ -36,14 +36,70 @@ mod sha2;
 mod workflow;
 
 // Whenever execution reaches somewhere it isn't supposed to rust code will "panic". Our panic
-// handler will print the available information on the screen. If we compile with `panic=abort`
-// this code will never get executed.
+// handler will print the available information on the screen and over RTT. If we compile with
+// `panic=abort` this code will never get executed.
 #[cfg(not(test))]
 #[cfg(not(feature = "testing"))]
 #[cfg_attr(feature = "bootloader", allow(unused_variables))]
 #[panic_handler]
 fn panic(info: &core::panic::PanicInfo) -> ! {
+    ::util::log::log!("{}", info);
     #[cfg(feature = "firmware")]
     bitbox02_rust::print_debug!(0, "Error: {}", info);
     loop {}
 }
+
+#[no_mangle]
+pub extern "C" fn rust_rtt_init() {
+    ::util::log::rtt_init();
+}
+
+// SAFETY: The pointer must point to a null terminated string
+#[no_mangle]
+#[cfg_attr(not(feature = "rtt"), allow(unused))]
+pub extern "C" fn rust_log(ptr: *const ::util::c_types::c_char) {
```

Comment:
If you use crate::util::Bytes here as the input argument this function can be safe.

Reply:
yeah, but I would say that would be less idiomatic C. When you pass strings in C they are typically null terminated and not ptr+len.

Reply:
True :+1:  You should mark the function `unsafe`.